spec/spec_enumerables.rb

```diff
@@ -0,0 +1,207 @@
+require './enumerable_methods'
+
+describe Enumerable do
+  let(:friends) { %w[Sharon Leo Leila Brian Arun Jean] }
+
+  describe 'my_each' do
+    it 'returns to_enum if no arguments or block were given' do
+      expect([1, 2, 3].my_each).to be_a Enumerator
+    end
+    it 'returns the array when a block is given' do
+      expect([1, 2, 3].my_each { |ele| }).to eql([1, 2, 3])
+    end
+  end
+
+  describe 'my_each_with_index' do
+    it 'returns to_enum if no arguments or block were given' do
+      expect([1, 2, 3].my_each_with_index).to be_a Enumerator
+    end
+    it 'returns an array with its index if block were given' do
+      expect([1, 2, 3].my_each_with_index { |ele, idx| }).to eql([1, 2, 3])
+    end
+  end
+
+  describe 'my_select' do
+    it 'returns an array containing all elements of enum for which the given block returns a true value.' do
+      expect([1, 2, 3, 4, 8].my_select(&:even?)).to eql([2, 4, 8])
+    end
+    it 'If no block is given, an Enumerator is returned instead' do
+      expect([1, 2, 3, 4].my_select).to be_a Enumerator
+    end
+  end
+
+  let(:array_of_strings) { %w[ant bear cat] }
+  let(:eg_range) { (1..15) }
+  describe '#my_all?' do
+    context 'when object is an array' do
+      it '#my_all?' do
```

Comment:
You are already telling that it is from `#my_all?` method you don't need to tell again that it is from the same method. Instead, just make it as you did before with previous tests. `describe '#my_all?' -> it 'when object is array'`

Do it with all the other test cases.